src/RejectedPromise.php

```diff
@@ -8,10 +8,10 @@ final class RejectedPromise implements PromiseInterface
 
     public function __construct($reason)
     {
-        if (!$reason instanceof \Throwable && !$reason instanceof \Exception) {
+        if (!$reason instanceof \Throwable) {
```

Comment:
Changes LGTM locally, but this can probably be replaced with a type hint now :+1: The `reject()` function can be adjusted likewise :+1:

Reply:
Done :tada: